dpctl/tensor/libtensor/include/kernels/reductions.hpp

```diff
@@ -111,7 +123,16 @@ struct SequentialReduction
             const ssize_t inp_offset = inp_iter_offset + inp_reduction_offset;
 
             using dpctl::tensor::type_utils::convert_impl;
-            outT val = convert_impl<outT, argT>(inp_[inp_offset]);
+            outT val;
+            if constexpr (std::is_same_v<ReductionOp,
+                                         sycl::logical_and<outT>> ||
+                          std::is_same_v<ReductionOp, sycl::logical_or<outT>>)
```

Comment:
Use `su_ns::IsLogicalOr<outT, ReductionOp>::value || su_ns::IsLogicalAnd<outT, ReductionOp>::value`.

Reply:
I think this same remark applies to other instances in this file.